src/store/inflight_activation.rs

```diff
@@ -659,7 +742,7 @@ impl InflightActivationStore {
             {
                 forwarder.to_discard.push(activation.id)
             } else if retry_state.on_attempts_exceeded == OnAttemptsExceeded::Deadletter as i32 {
-                forwarder.to_deadletter.push(activation)
+                forwarder.to_deadletter.push(activation.id)
```

Comment:
Would we be able to revise this logic to not need to deserialize the activation in a future change?

Reply:
Oops yeah I had that but it got merged out. Fixed.